### PR TITLE
PLAT-75973: Add support for <count>

### PR DESCRIPTION
**Pull request by ryanjduffy**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Ability to count the number of items in a list

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add a `<count>` pseudo-attribute which returns the number of nodes that match `selector`

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
This will not adequately address virtualized lists which should instead rely on the `aria-setsize` attribute set on list items.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -144,18 +144,28 @@ const matchesRules = (ruleset, msg) => Object.keys(ruleset).some(key => {
...
 
+const getFirstNode = (nodeOrList) => {
+	return nodeOrList instanceof global.HTMLElement ? nodeOrList : nodeOrList[0];
+};
+
 const resolveAttribute = (name) => (node) => {
-	if (!node) return null;
+	if (!node || node.length === 0) return null;
+
+	if (name === '<count>') {
```
Comment (webOS101) [inline index.js:159]: Is there someplace to document this special attribute?

Comment (ryanjduffy) [reply index.js:159]: Working on it now

**Comment by ryanjduffy** [comment]
I added docs for `'<count>'` and `'<value>'` in the temporary DSL docs but we'll need to refine that when the full JSDocs are added.

I also refined the data resolution logic a bit for `'<count>'` so that when it is used as the `value` but the targeted node isn't found, it will return `0`. So if a resolver targets a node using `closest` or `selector` with `value: '<count>'` and the node isn't found, it will return `0`. However, if a resolver uses the same targeting but `value` is an object (even if that object only includes `value: '<count>'`, the data will be omitted as was currently the case for unmatched resolvers.

To illustrate:

```js
data: {
  equalsZero: { // will equal 0
    closest: 'does-not-exist',
    value: '<count>'
  },
  notDefined1: { // will be undefined
    closest: 'does-not-exist',
    value: '<text>'
  },
  notDefined2: { // will be undefined
    closest: 'does-not-exist',
    value: {
      selector: 'button',
      value: '<count>'
    }
  },
  notDefined3: { // will be undefined
    closest: 'does-not-exist',
    value: {
      value: '<count>'
    }
  }
}
```

Open to discussion if this doesn't make sense.

**Review by JayCanuck — APPROVED** [review]
LGTM :+1:

**Pull request merged**